test/python/test_layout.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+""" Tests the layout object"""
+
+import unittest
+
+from qiskit.mapper import Layout
+from .common import QiskitTestCase
+
+
+class LayoutTest(QiskitTestCase):
+    """Test the methods in the layout object."""
+
+    def test_layout_from_dict(self):
+        """Constructor from a dict"""
+        layout = Layout({('qr', 0): 0,
+                         ('qr', 1): 1,
+                         ('qr', 2): 2})
+
+        self.assertEqual(layout[('qr', 0)], 0)
+        self.assertEqual(layout[('qr', 1)], 1)
+        self.assertEqual(layout[('qr', 2)], 2)
+        self.assertEqual(layout[0], ('qr', 0))
+        self.assertEqual(layout[1], ('qr', 1))
+        self.assertEqual(layout[2], ('qr', 2))
+
+    def test_layout_from_list(self):
+        """Constructor from a list"""
+        layout = Layout([('q0', 0), ('q1', 0), None, ('q1', 1), ('q0', 1)])
+
+        self.assertEqual(layout[('q0', 0)], 0)
+        self.assertEqual(layout[('q1', 0)], 1)
+        self.assertEqual(layout[None], None)
```

Comment:
Shouldn't this raise an error? The None is not an item in the layout. It is the lack of an item. And you could have multiple Nones in the list (multiple physical bits that nothing is assigned to).